### DISCUSSION TITLE
Can you use view/edit buttons with code hosted in Gitea?

### DISCUSSION THREAD

**Discussion by macpijan** [squidfunk/mkdocs-material#6900] [Q&A]
You can configure the edit_uri via: https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/?h=action+view#code-actions

It seems that the same pattern is used for `action.view`, with the assumption that `raw` is used in place of `edit`.

This works for GitHub, but for Gitea the links are as follows:

https://XXX/_edit/master/docs/doc.md
https://XXX/raw/branch/master/doc.md

Perhaps there is another way to configure, or it deserves a feature request?

**Comment by kamilkrzyskow** [comment]
Hello @macpijan,
the `edit_uri` setting is the sole config option related to this provided by MkDocs: https://www.mkdocs.org/user-guide/configuration/#edit_uri
~~It's unlikely they will add a separate `raw_uri` option.~~ Actually I don't know if it's unlikely 😅, they just have a lot of issues etc., so another feature request will be a pebble in a pond. Don't let me discourage you from making suggestions 👌 

The theme replaces the `edit` with `raw` like you said in the template:
https://github.com/squidfunk/mkdocs-material/blob/c8bae9b5b4b46f0032a9f2dddb30adda68158c78/src/templates/partials/actions.html#L40-L46

I'm not sure what kind of feature request you want to propose, but you can modify it with customization:

- https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme

override the [actions.html](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/actions.html) partial and change the logic to the one you want.

---

Another option would be to replace the address with extra JavaScript: 
- https://squidfunk.github.io/mkdocs-material/customization/#additional-javascript

This discussion contains a lot of reference about working with the action buttons from JavaScript, it focuses on adding one, but replacing a href should be even simpler.
- https://github.com/squidfunk/mkdocs-material/discussions/6849
